### ISSUE TITLE
loadState overwritten for each request

### ISSUE THREAD

**Issue by kjanoudi** [Rezonans/redux-async-connect#50]
If i use reduxAsyncConnect in multiple components that are being rendered, loadState is overwritten in some situations (race condition). Rather than replace loadState as is being done now:

``` js
case LOAD:
      return {
        ...state,
        loadState: {
          [action.key]: {
            loading: true,
            loaded: false
          }
        }
      };

```

i suggest merging loadStates and only replacing the keys that match.

**Comment by AVVS** [comment]
Js is one threaded, therefore loop is blocked and 2 "async" actions won't ever call reducers with the same state twice

**Comment by kjanoudi** [comment]
@AVVS while that's true, it seems that by not using the Object.assign method of setting the new state, the loadState key is replaced fully. So you will never be able to have:

``` js
state: {
  loadState: {
     config: { loading: true, loaded: false},
     users: {loaded: true, loading: false}
  }
}

```

instead only able to have _either_ loadstate.config _or_ loadstate.users but never both

rather than returning this:

``` js
return {
        ...state,
        loadState: {
          [action.key]: {
            loading: true,
            loaded: false
          }
        }
      };
```

i think this would solve it:

``` js
return _.merge({}, state, {
            loadState: {
               [action.key]: {
                   loading: true,
                  loaded: false
               }
            }
        })
```

**Comment by AVVS** [comment]
Thats true. But its not a race condition, just lack of deep merge. Should do spread assign to this property as well and that would solve it

**Comment by sars** [comment]
@kjanoudi , yes, looks like a bug.
Any PR or help appreciated. 
Or i'll fix it by myself and a little bit later
